tests/src/system/basic/WskBasicTests.scala

```diff
@@ -291,6 +292,72 @@ class WskBasicTests
+            stdout should include regex (""""key":\s*"object",\s*"value":\s*\{\s*"objBool":\s*false,\s*"objStrNum":\s*"123",\s*"objString":\s*"aString",\s*"objStrArr":\s*\[\s*"1",\s*"2"\s*\],\s*"objNum":\s*300,\s*"objNumArr":\s*\[\s*1,\s*2\s*\]\s*\}""")
+    }
+
+    it should "create, and get a package to verify parameter parsing" in withAssetCleaner(wskprops) {
+        (wp, assetHelper) =>
+            val name = "packageParameters"
+
+            assetHelper.withCleaner(wsk.pkg, name) {
+                (pkg, _) =>
+                    pkg.create(name, parameters = getJSONTestArguments)
+            }
+            val stdout = wsk.pkg.get(name).stdout
+
+            stdout should include regex (""""key":\s*"string",\s*"value":\s*"This is a string"""")
+            stdout should include regex (""""key":\s*"strNum",\s*"value":\s*"9"""")
+            stdout should include regex (""""key":\s*"number",\s*"value":\s*8""")
+            stdout should include regex (""""key":\s*"numArr",\s*"value":\s*\[\s*44,\s*55\s*\]""")
+            stdout should include regex (""""key":\s*"strArr",\s*"value":\s*\[\s*"44",\s*"55"\s*\]""")
+            stdout should include regex (""""key":\s*"objArr",\s*"value":\s*\[\s*\{\s*"name":\s*"someName",\s*"required":\s*true\s*\},\s*\{\s*"name":\s*"events",\s*"count":\s*10\s*\}\s*\]""")
+            stdout should include regex (""""key":\s*"object",\s*"value":\s*\{\s*"objBool":\s*false,\s*"objStrNum":\s*"123",\s*"objString":\s*"aString",\s*"objStrArr":\s*\[\s*"1",\s*"2"\s*\],\s*"objNum":\s*300,\s*"objNumArr":\s*\[\s*1,\s*2\s*\]\s*\}""")
+    }
+
+    it should "not create a package when -a is specified without arguments" in withAssetCleaner(wskprops) {
+        (wp, assetHelper) =>
+            val stderr = wsk.cli(wskprops.overrides ++ Seq("package", "create", "packageName", "--auth", wp.authKey,
+                "-a"), expectedExitCode = ERROR_EXIT).stderr
+            stderr should include("Annotation arguments must be a key value pair")
+    }
+
+    it should "not create a package when -p is specified without arguments" in withAssetCleaner(wskprops) {
+        (wp, assetHelper) =>
+            val stderr = wsk.cli(wskprops.overrides ++ Seq("package", "create", "packageName", "--auth", wp.authKey,
+                "-p"), expectedExitCode = ERROR_EXIT).stderr
+            stderr should include("Parameter arguments must be a key value pair")
+    }
+
+    it should "not create a package when an annotation contains invalid JSON" in withAssetCleaner(wskprops) {
+        (wp, assetHelper) =>
+            val stderr = wsk.cli(wskprops.overrides ++ Seq("package", "create", "packageName", "--auth", wp.authKey,
+                "-a", "key", "{\"invalidObj\", }"), expectedExitCode = ERROR_EXIT).stderr
+            stderr should include("Error encoding request body")
```

Comment:
Customer facing error message is a bit cryptic.  Is it possible to have an error message that is more clear (i.e. "The annotation value is not valid JSON: <json encoding error details>")?